src/Compilers/CSharp/Portable/FlowAnalysis/DefiniteAssignment.cs

```diff
@@ -1690,16 +1690,30 @@ protected override void EnterParameter(ParameterSymbol parameter)
                 sourceComplexParam.BindParameterAttributes() is { IsDefaultOrEmpty: false } boundAttributes)
             {
                 // Mark attribute arguments as used.
-                foreach (var boundAttribute in boundAttributes)
+                VisitAttributes(boundAttributes);
+            }
+        }
+
+        /// <summary>
+        /// Marks attribute arguments as used.
+        /// </summary>
+        private void VisitAttributes(ImmutableArray<(CSharpAttributeData, BoundAttribute)> boundAttributes)
+        {
+            foreach (var (attributeData, boundAttribute) in boundAttributes)
```

Comment:
nit: might it make sense to move the check for `IsDefault` here, so that the `{ IsDefaultOrEmpty: false }` checks below could be factored away?

Reply:
Good catch, thanks. 199920676bedcc3ac63467eda2c7974cfe74585d